src/Servers/Kestrel/stress/Program.cs

```diff
@@ -0,0 +1,750 @@
+            }
+            return;
+        }
+
+        if (opIndices != null)
+        {
+            clientOperations = opIndices.Select(i => clientOperations[i]).ToArray();
+        }
+
+        Console.WriteLine("     .NET Core: " + Path.GetFileName(Path.GetDirectoryName(typeof(object).Assembly.Location)));
+        Console.WriteLine("  ASP.NET Core: " + Path.GetFileName(Path.GetDirectoryName(typeof(WebHost).Assembly.Location)));
+        Console.WriteLine("       Tracing: " + (logPath == null ? (object)false : logPath.Length == 0 ? (object)true : logPath));
+        Console.WriteLine("   ASP.NET Log: " + aspnetLog);
+        Console.WriteLine("   Concurrency: " + concurrentRequests);
+        Console.WriteLine("Content Length: " + maxContentLength);
+        Console.WriteLine(" HTTP Versions: " + string.Join<Version>(", ", httpVersions));
+        Console.WriteLine("      Lifetime: " + (connectionLifetime.HasValue ? $"{connectionLifetime}ms" : "(infinite)"));
+        Console.WriteLine("    Operations: " + string.Join(", ", clientOperations.Select(o => o.Item1)));
+        Console.WriteLine("   Random Seed: " + seed);
+        Console.WriteLine();
+
+        // Start the Kestrel web server in-proc.
+        Console.WriteLine("Starting server.");
+        WebHost.CreateDefaultBuilder()
+
+            //Use Kestrel, and configure it for HTTPS with a self - signed test certificate.
+            .UseKestrel(ko =>
+            {
+                ko.ListenLocalhost(HttpsPort, listenOptions =>
+                {
+                    using (RSA rsa = RSA.Create())
+                    {
+                        var certReq = new CertificateRequest($"CN={LocalhostName}", rsa, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+                        certReq.CertificateExtensions.Add(new X509BasicConstraintsExtension(false, false, 0, false));
+                        certReq.CertificateExtensions.Add(new X509EnhancedKeyUsageExtension(new OidCollection { new Oid("1.3.6.1.5.5.7.3.1") }, false));
+                        certReq.CertificateExtensions.Add(new X509KeyUsageExtension(X509KeyUsageFlags.DigitalSignature, false));
+                        X509Certificate2 cert = certReq.CreateSelfSigned(DateTimeOffset.UtcNow.AddMonths(-1), DateTimeOffset.UtcNow.AddMonths(1));
+                        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        {
+                            cert = new X509Certificate2(cert.Export(X509ContentType.Pfx));
+                        }
+                        listenOptions.UseHttps(cert);
+                    }
+                });
+            })
+
+            // Output only warnings and errors from Kestrel
+            .ConfigureLogging(log => log.AddFilter("Microsoft.AspNetCore", level => aspnetLog ? level >= LogLevel.Error : false))
+
+            // Set up how each request should be handled by the server.
+            .Configure(app =>
```

Comment:
This is substantial enough to warrant a Startup class or at least its own method.